index.html

```diff
@@ -3,134 +3,453 @@
+    svg2.append("g")
+      .attr("class","y axis")
+      .attr("transform","translate(0,"+ height + ")")
+      .call(xAxis2);
+
+    svg2.append("g")
+      .attr("class","y axis")
+      .attr("transform","translate("+ padding + ",0)")
+      .call(yAxis2)
+      ;
+
+    svg2.append("text")
+      .attr("text-anchor", "middle")  // this makes it easy to centre the text as the transform is applied to the anchor
+      .attr("class","axislabel")
+      .attr("transform", "translate("+ (-axispadding) +","+(height/2)+")rotate(-90)")  // text is drawn off the screen top left, move down and out and rotate
+      .text("Arrests per 100,000 people");
+
+    svg2.append("text")
+      .attr("text-anchor", "middle")  // this makes it easy to centre the text as the transform is applied to the anchor
+      .attr("class","axislabel")
+      .attr("transform", "translate("+ (width/2) +","+(height+(xaxispadding))+")")  // centre below axis
+      .text("Median Income");
+
+    // EVENT HANDLERS
     d3.select('#year-range').on('change', function () {
-      options.yearRange = d3.event.target.value;
+      options.state = d3.event.target.value;
+      options.region = 'all';
+      console.log(xAxis.ticks());
+      update();
+    });
+
+    d3.select('#region-select').on('change', function () {
+      options.region = d3.event.target.value;
+      options.state = 'all';
       update();
     });
   }
 
+  function customAxis(g) {
+  g.selectAll("text")
+      .attr("x", 4)
+      .attr("dy", -4);
+}
+
   function update() {
     // TRANSFORM
     txData = data.slice();
-
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
+    if (options.state !== 'all') {
+      var state = options.state;
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+        return d.Abbr === state;
       });
     }
-
+    if (options.region !== 'all') {
+      var region = options.region;
+      txData = txData.filter(function (d) {
+        return d.Region === region;
+      });
+    }
+    
     // SCATTERPLOT
+    var points = svg.selectAll('.point')
+      .data(txData);
+
+    points.enter().append('circle')
+      .attr('class', 'point');
+
+    points
+      .attr('cx', function (d) { return x(d.PovertyRate/100); })
+      .attr('cy', function (d) { return y(d.rate); })
+      .attr('r', function (d) { return rscale(d.Pop); });
+    points.exit().remove();
+
+    //2nd Scatterplot
+     var points2 = svg2.selectAll('.point2')
+      .data(txData);
 
-    // HISTOGRAMS
+    points2.enter().append('circle')
+      .attr('class', 'point2');
+
+    points2
+      .attr('cx', function (d) { return x2(d.MedianIncome); })
+      .attr('cy', function (d) { return y(d.rate); })
+      .attr('r', function (d) { return rscale(d.Pop); });
+    points2.exit().remove();
+
+
+
+      // HISTOGRAMS
+     var bins = d3.layout.histogram()
+      .bins(x.ticks().length-1)
+      (txData);
```

Comment:
So you've probably figured out what's missing here. Refer to the class demonstration of `histogram.value()` or use `.map()` on your data to make sure the histogram has access to the specific column of data you're using.